core/domain/stats_jobs_one_off.py

```diff
@@ -303,6 +323,39 @@ def reduce(exp_id, values):
 
     @classmethod
     def prepare_reduce(cls, exp_id, values):
+        """Runs the reduce step to extract aggregate exploration statistics from
+        the dicts passed in by the mapping function.
+
+        Args:
+            exp_id: str. The exploration id.
+            values: list(str). List of events for preparing the reduce.
+
+        Returns:
+            tuple(list(dict), list(ExplorationStats), list(str)). 3-tuple where:
+            - The first element is the list of dicts of different version-wise
+                ExplorationStats. Each dict contains the following key-value
+                pairs:
+                - num_starts_v1: int. Number of learners who started the
+                    exploration.
+                - num_starts_v2: int. As above, but for events with version 2.
+                - num_actual_starts_v1: int. Number of learners who actually
+                    attempted the exploration. These are the learners who have
+                    completed the initial state of the exploration and traversed
+                    to the next state.
+                - num_actual_starts_v2: int. As above, but for events with
+                    version 2.
+                - num_completions_v1: int. Number of learners who completed the
+                    exploration.
+                - num_completions_v2: int. As above, but for events with
+                    version 2.
+                - state_stats_mapping: dict. A dictionary mapping the state
+                    names of an exploration to the corresponding StateStats
+                    domain object.
+            - The second element is the list of ExplorationStats domain class
```

Comment:
Check: is this domain object or dict?

Reply:
It's a list of `ExplorationStats` domain objects as stated in Returns (verified after printing as well).